rust/kernel/sync/mutex.rs

```diff
@@ -0,0 +1,95 @@
+// SPDX-License-Identifier: GPL-2.0
+
+use super::{Guard, Lock};
+use crate::{bindings, CStr};
+use core::{cell::UnsafeCell, pin::Pin};
+
+/// Safely initialises a [`Mutex`] wih the given name and new lock class.
```

Comment:
```suggestion
/// Safely initialises a [`Mutex`] with the given name, generating a new lock class.
```

i.e. it sounded like the new lock class is also passed rather than generated inside.

Reply:
Done.